From fdae8096fe71e65f7c13575c990eb2652a328122 Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@linux-m68k.org>
Date: Thu, 19 Aug 2010 07:42:16 -0700
Subject: [PATCH] Fix ifunc thunk for strspn on x86 in static libc
 (cherry picked from commit bdaa77bc20bbaf5a78da541b993cfead8b8c1b1c)

---
 ChangeLog                            |    4 ++++
 sysdeps/i386/i686/multiarch/strspn.S |    4 ++--
 2 files changed, 6 insertions(+), 2 deletions(-)

2010-08-19  Andreas Schwab  <schwab@redhat.com>

	* sysdeps/i386/i686/multiarch/strspn.S [!SHARED]: Fix SSE4.2 check.

diff --git a/sysdeps/i386/i686/multiarch/strspn.S b/sysdeps/i386/i686/multiarch/strspn.S
index 7fe6f97..dbdf1af 100644
--- a/sysdeps/i386/i686/multiarch/strspn.S
+++ b/sysdeps/i386/i686/multiarch/strspn.S
@@ -1,5 +1,5 @@
 /* Multiple versions of strspn
-   Copyright (C) 2009 Free Software Foundation, Inc.
+   Copyright (C) 2009,2010 Free Software Foundation, Inc.
    Contributed by Intel Corporation.
    This file is part of the GNU C Library.
 
@@ -65,7 +65,7 @@ ENTRY(strspn)
 	jne	1f
 	call	__init_cpu_features
 1:	leal	__strspn_ia32, %eax
-	testl	$index_SSE2, CPUID_OFFSET+index_SSE4_2+__cpu_features
+	testl	$bit_SSE4_2, CPUID_OFFSET+index_SSE4_2+__cpu_features
 	jz	2f
 	leal	__strspn_sse42, %eax
 2:	ret
-- 
1.7.2

